mep/books/models.py

```diff
@@ -127,7 +127,62 @@ def create_from_uri(cls, uri):
                        uri, response.status_code)
 
 
-class Work(Notable, Indexable):
+class WorkSignalHandlers:
```

Comment:
if you think it makes sense, I think this could be an associated class (like django's `Meta`) of the model itself, since it doesn't really have meaning outside of the model. then each model would optionally have a class called `SignalHandlers`.

Reply:
although I guess convention for extending `QuerySet`s is that they are an entire class separate from the model, so maybe it wouldn't be consistent to do this?

Reply:
maybe whenever we get around to writing "smart" stuff that can derive handlers automatically, it will work like a `QuerySet`, where you get some behavior by default and can create a custom class/extend it if you need more behavior.

Reply:
Maybe an `IndexDependencyManager` or something like that? I think the handlers are all very similar: find and index the set of impacted objects based on some relationship to the current instance that caused the signal to be fired. Variations on the delete handling depending on whether a foreign key (post_save) or many to many (pre_save and needs to call clear). I think it should be possible to refactor into some generic handling, and now we have a lot of examples to work from.

Reply:
Love the idea of calling it a `Manager` too.